x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/TransformTests.java

```diff
@@ -36,6 +50,9 @@ public void testNodeAttributes() {
             transformEnabled,
             Boolean.parseBoolean(transform.additionalSettings().get("node.attr.transform.node"))
         );
+        if (transformPluginEnabled && useExplicitSetting && useLegacySetting) {
```

Comment:
I suggest to incorporate the test fix from #58525, which is just this one line change:
```suggestion
        if (useExplicitSetting && useLegacySetting) {
```